### DISCUSSION TITLE
addConsumptionOffset API Command Question

### DISCUSSION THREAD

**Discussion by ChutneyMary** [ngardiner/TWCManager#544] [General]
I'm making some modifications to my automation system and wanted to clarify the behaviour of the addConsumptionOffset API.

I can choose to feed just negative values for 'excess' energy into addConsumptionOffset, or I can have a continuous supply of positive to negative values for the household energy state.

I wanted to clarify whether the addConsumptionOffset has a 'PID' (Proportional, Integral, Derivative) type control or is it much simpler.

ie. two scenarios -

1. I'm feeding -16 amps to addConsumptionOffset and a cloud comes over and the household energy state swings to a large consumption value (+8). Will this command a quicker swing to not charging?
2. I'm feeding -16 amps to addConsumptionOffset and a cloud comes over and the household energy state swings to consumption (0 to addConsumptionOffset). Will this command be the same rate as above?

**Comment by MikeBishop** [comment]
It's very simple. It's treated as another source of consumption/generation data that always returns a fixed amount when checked. TWCManager sums all the consumption/generation available in the system, though most people only have a single source that returns non-zero values.

If you're in a Track Green Energy policy, there is some logic that doesn't immediately stop charging (just drops down to the TWC's minimum amperage) to see if power becomes available again.
